src/modules/Elsa.Workflows.Runtime/Services/WorkflowDefinitionRefresher.cs

```diff
@@ -0,0 +1,55 @@
+using Elsa.Common.Entities;
+using Elsa.Common.Models;
+using Elsa.Mediator.Contracts;
+using Elsa.Workflows.Management.Contracts;
+using Elsa.Workflows.Management.Entities;
+using Elsa.Workflows.Management.Filters;
+using Elsa.Workflows.Runtime.Contracts;
+using Elsa.Workflows.Runtime.Notifications;
+using Elsa.Workflows.Runtime.Requests;
+using Elsa.Workflows.Runtime.Responses;
+
+namespace Elsa.Workflows.Runtime.Services;
+
+/// <inheritdoc />
+public class WorkflowDefinitionsRefresher(IWorkflowDefinitionStore store, ITriggerIndexer triggerIndexer, INotificationSender notificationSender) : IWorkflowDefinitionsRefresher
+{
+    /// <inheritdoc />
+    public async Task<RefreshWorkflowDefinitionsResponse> RefreshWorkflowDefinitionsAsync(RefreshWorkflowDefinitionsRequest request, CancellationToken cancellationToken)
+    {
+        var filter = new WorkflowDefinitionFilter
+        {
+            DefinitionIds = request.DefinitionIds,
+            VersionOptions = VersionOptions.Published
+        };
+
+        var currentPage = 0;
+        var processedWorkflowDefinitions = new List<WorkflowDefinition>();
+        var batchSize = request.BatchSize;
+        var order = new WorkflowDefinitionOrder<string>(x => x.Id, OrderDirection.Ascending);
+
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            var pageArgs = PageArgs.FromPage(currentPage, batchSize);
+            var definitions = await store.FindManyAsync(filter, order, pageArgs, cancellationToken);
+
+            if (definitions.Items.Count == 0)
```

Comment:
We can also add a check for count < batchSize. That will prevent an extra call to the database.

Reply:
Yes, nice. @MariusVuscanNx can we incorporate that in your next PR?